### PR TITLE
Update deployments for Kubernetes v1.8+

### LINKED ISSUE #1764: Deployment of Heketi and GlusterFS on Kubernetes
### Kind of issue

 Bug

### Observed behavior

During installation of heketi with glusterfs on Kubernetes, manifests are outdated and do not work correctly with Kubernetes version 1.8 without manipulation.

I have followed the following guides to install heketi and glusterfs on K8s:
https://www.cnblogs.com/ssgeek/p/11725648.html  (use google translate to translate to English if needed)
https://medium.com/faun/configuring-ha-kubernetes-cluster-on-bare-metal-servers-with-glusterfs-metallb-2-3-c9e0b705aa3d

The following errors exist:

1.  https://github.com/heketi/heketi/blob/master/extras/kubernetes/glusterfs-daemonset.json
error: unable to recognize "glusterfs-daemonset.json": no matches for kind "DaemonSet" in version "extensions/v1beta1"

Changes Required:
```
"apiVersion": "apps/v1"
```
and
```
"spec": {
    "selector": {
        "matchLabels": {
            "glusterfs-node": "daemonset"
        }
    }
```

2. https://github.com/heketi/heketi/blob/master/extras/kubernetes/heketi-bootstrap.json

Changes Required: 
```
      "kind": "Deployment",
      "apiVersion": "apps/v1"
```
and
```
      "spec": {
        "selector": {
          "matchLabels": {
            "name": "deploy-heketi" 
          }
        },
```

### Expected/desired behavior

Installation of heketi without issue on Kubernetes version 1.8

### Details on how to reproduce (minimal and precise)

Run installation steps as normal on Kubernetes 1.8

### Information about the environment:

- Heketi version used : https://github.com/heketi/heketi/tree/v9.0.0 (https://github.com/heketi/heketi/releases/download/v9.0.0/heketi-client-v9.0.0.linux.amd64.tar.gz)
- Operating system used: Centos 7
- Heketi compiled from sources, as a package (rpm/deb), or container: Version 9.0.0  from link above
- If container, which container image: 
- Using kubernetes, openshift, or direct install:   kubernetes
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside:   gluster running inside kubernetes
- If kubernetes/openshift, how was it deployed (gk-deploy, openshift-ansible, other, custom):  using guides above.

### Other useful information

> kubectl get all

```
NAME                          READY   STATUS    RESTARTS   AGE
pod/glusterfs-56w6w           1/1     Running   1          12d
pod/glusterfs-vnj7c           1/1     Running   0          19d
pod/glusterfs-wwhrv           1/1     Running   2          19d
pod/heketi-77df87b55d-nt7c5   1/1     Running   2161       8d

NAME                                                             TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
service/glusterfs-dynamic-11dd4063-790a-4627-854c-32d7d8ce820c   ClusterIP   10.110.219.158   <none>        1/TCP      14d
service/glusterfs-dynamic-d14611d0-b497-495c-ae51-88e7a9417755   ClusterIP   10.101.163.94    <none>        1/TCP      12d
service/glusterfs-dynamic-d31721c7-c132-4fbc-a6c4-f07774316363   ClusterIP   10.106.100.90    <none>        1/TCP      19d
service/glusterfs-dynamic-d6d7df2a-b994-4487-82c3-a6bd1bbbc0ca   ClusterIP   10.98.6.117      <none>        1/TCP      12d
service/glusterfs-dynamic-f0036164-8d34-4858-bcfa-875167deca35   ClusterIP   10.105.93.46     <none>        1/TCP      19d
service/heketi                                                   ClusterIP   10.105.28.227    <none>        8080/TCP   19d
service/heketi-storage-endpoints                                 ClusterIP   10.104.182.120   <none>        1/TCP      19d
service/kubernetes                                               ClusterIP   10.96.0.1        <none>        443/TCP    30d

NAME                       DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR           AGE
daemonset.apps/glusterfs   3         3         3       3            3           storagenode=glusterfs   19d

NAME                     READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/heketi   1/1     1            1           19d

NAME                                DESIRED   CURRENT   READY   AGE
replicaset.apps/heketi-77df87b55d   1         1         1       19d
```

kubectl get nodes -o wide
```
NAME     STATUS   ROLES    AGE   VERSION   INTERNAL-IP    EXTERNAL-IP   OS-IMAGE                KERNEL-VERSION                CONTAINER-RUNTIME
davila   Ready    master   27d   v1.18.6   172.20.46.59   <none>        CentOS Linux 7 (Core)   3.10.0-1127.18.2.el7.x86_64   docker://19.3.12
flatro   Ready    master   30d   v1.18.6   172.20.46.60   <none>        CentOS Linux 7 (Core)   3.10.0-1127.18.2.el7.x86_64   docker://19.3.12
roccas   Ready    master   30d   v1.18.6   172.20.46.61   <none>        CentOS Linux 7 (Core)   3.10.0-1127.13.1.el7.x86_64   docker://19.3.12
```

### PR DESCRIPTION
**Pull request by AnthonyPluth**
### What does this PR achieve? Why do we need it?
Fixes deployment issues on kubernetes versions that don't support `extensions/v1beta1` (versions 1.8+)

### Does this PR fix issues?

Fixes #1764

### CODE REVIEW

**Comment by raghavendra-talur** [comment]
@AnthonyPluth I understand that the changes which replace "extensions/v1beta1" with "apps/v1". What is the reasoning behind addition of label selectors?

**Comment by AnthonyPluth** [comment]
@raghavendra-talur labels are required in deployments when using apps/v1

File: extras/kubernetes/heketi-bootstrap.json [hunk 1]
```diff
@@ -42,6 +42,11 @@
         }
       },
       "spec": {
+        "selector": {
+          "matchLabels": {
+            "name": "deploy-heketi"
```
Comment (raghavendra-talur) [inline extras/kubernetes/heketi-bootstrap.json:47]: I see that you have copied only one of the three labels in the template. Any particular reason for that?

**Comment by raghavendra-talur** [comment]
> @raghavendra-talur labels are required in deployments when using apps/v1

Thanks! Here is the link for future reference: https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/#pod-selector

**Review by raghavendra-talur — APPROVED** [review]

**Review by phlogistonjohn — APPROVED** [review]

**Pull request merged**